core/src/main/scala/handler/MethodLabelHandler.scala

```diff
@@ -0,0 +1,54 @@
+/** *
+  * Copyright 2014 Rackspace US, Inc.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+package com.rackspace.com.papi.components.checker.handler
+
+import javax.servlet.FilterChain
+
+import com.rackspace.com.papi.components.checker.Validator
+import com.rackspace.com.papi.components.checker.servlet.{CheckerServletRequest, CheckerServletResponse}
+import com.rackspace.com.papi.components.checker.step.{StepContext, Step, Result}
+import org.w3c.dom.{Document, NodeList}
+
+class MethodLabelHandler extends ResultHandler  {
+  val MethodLabelHeaderName: String = "X-METHOD-LABEL"
+
+  var methodMap : Map[String, String] = Map()
+
+  override def init(validator: Validator, checker: Option[Document]): Unit = {
+    checker.map { doc =>
+      val steps: NodeList = doc.getElementsByTagName("step")
+      for {
+        i <- 0 until steps.getLength
+        item <- Option(steps.item(i))
+        attributes <- Option(item.getAttributes)
+        attType <- Option(attributes.getNamedItem("type"))
+        id <- Option(attributes.getNamedItem("id")) if "METHOD".equalsIgnoreCase(attType.getTextContent)
+        label <- Option(attributes.getNamedItem("label")) if "METHOD".equalsIgnoreCase(attType.getTextContent)
+        idText <- Option(id.getTextContent)
+        labelText <- Option(label.getTextContent)
+      } yield {
+        methodMap = methodMap ++ Map(idText -> labelText)
+      }
+    }
+  }
+
+  override def handle(req: CheckerServletRequest, resp: CheckerServletResponse, chain: FilterChain, result: Result): Unit = {}
+
+  override def inStep (currentStep: Step, req: CheckerServletRequest, resp : CheckerServletResponse, context: StepContext) : Unit = {
+    methodMap.get(currentStep.id).map(req.addHeader("X-METHOD-LABEL",_))
```

Comment:
This won't work 😞

You'll end up setting multiple headers in some cases.  The reason for this is that it is possible for a resource to have multiple ```POST``` methods each with its own label.  There are a lot of examples of this....the easiest one to explain is Atom Hopper.  You can POST a load balancer create message or you can POST a load balancer delete message to the same feed  -- what sets the methods apart is a property of the content which is checked *after* the method is checked.

Instead of adding a header, you should create a new context with the result header.  If the path isn't followed trough -- that context will pop out of the stack.